[WPE][GTK] libsysprof-capture contains GPLv3+ sources

mcatanzaro opened the pull request:
#### e4ef1b87b83982ec61e904a029ff8540096afe51
<pre>
[WPE][GTK] libsysprof-capture contains GPLv3+ sources
<a href="https://bugs.webkit.org/show_bug.cgi?id=286273">https://bugs.webkit.org/show_bug.cgi?id=286273</a>

Reviewed by Adrian Perez de Castro.

Update mapped-ring-buffer.c and mapped-ring-buffer.h to the latest
upstream versions, to fix the licensing.

Also, delete the tests. Several of them are GPLv3+, and they are not
even built so there&apos;s no need for them.

* Source/ThirdParty/libsysprof-capture/README.WebKit:
* Source/ThirdParty/libsysprof-capture/mapped-ring-buffer.c:
* Source/ThirdParty/libsysprof-capture/mapped-ring-buffer.h:
* Source/ThirdParty/libsysprof-capture/tests/allocs-by-size.c: Removed.
* Source/ThirdParty/libsysprof-capture/tests/cross-thread-frees.c: Removed.
* Source/ThirdParty/libsysprof-capture/tests/find-temp-allocs.c: Removed.
* Source/ThirdParty/libsysprof-capture/tests/meson.build: Removed.
* Source/ThirdParty/libsysprof-capture/tests/rewrite-pid.c: Removed.
* Source/ThirdParty/libsysprof-capture/tests/test-capture-cursor.c: Removed.
* Source/ThirdParty/libsysprof-capture/tests/test-capture.c: Removed.
* Source/ThirdParty/libsysprof-capture/tests/test-cplusplus.cpp: Removed.
* Source/ThirdParty/libsysprof-capture/tests/test-mapped-ring-buffer.c: Removed.

Canonical link: <a href="https://commits.webkit.org/289297@main">https://commits.webkit.org/289297@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/8a0c17e98acea44e93d1f82fd0c05426720e5126

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/85579 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/5275 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/39982 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/90670 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/36583 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/5473 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/13255 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/66506 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/24321 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/88592 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/4201 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/77705 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/46789 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/4070 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/31968 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/35653 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/74739 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/32824 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/92331 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/12901 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/9462 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/75225 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/13116 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/73538 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/74362 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/18418 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/18594 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/17037 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/5019 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/12920 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/12720 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/16153 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/14501 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

aperezdc commented on Source/ThirdParty/libsysprof-capture/README.WebKit:
```diff
@@ -13,6 +13,7 @@ from scratch, and restoring the needed files with Git. Example commands:
       Source/ThirdParty/libsysprof-capture/README.WebKit \
       Source/ThirdParty/libsysprof-capture/CMakeLists.txt \
       Source/ThirdParty/libsysprof-capture/include/config.h
+  git rm Source/ThirdParty/libsysprof-capture/tests
```
I think we would prefer that the working copy doesn't have stray files. Is there some reason to not use `rm -rf` here?

mcatanzaro replied:
`git rm -r` deletes the files. I just forgot the -r flag.

mcatanzaro replied:
(`git rm` doesn't just remove files from the index. It also deletes them. It's equivalent to `rm` and then `git add`.)

mcatanzaro committed:
[WPE][GTK] libsysprof-capture contains GPLv3+ sources
https://bugs.webkit.org/show_bug.cgi?id=286273

Reviewed by Adrian Perez de Castro.

Update mapped-ring-buffer.c and mapped-ring-buffer.h to the latest
upstream versions, to fix the licensing.

Also, delete the tests. Several of them are GPLv3+, and they are not
even built so there's no need for them.

* Source/ThirdParty/libsysprof-capture/README.WebKit:
* Source/ThirdParty/libsysprof-capture/mapped-ring-buffer.c:
* Source/ThirdParty/libsysprof-capture/mapped-ring-buffer.h:
* Source/ThirdParty/libsysprof-capture/tests/allocs-by-size.c: Removed.
* Source/ThirdParty/libsysprof-capture/tests/cross-thread-frees.c: Removed.
* Source/ThirdParty/libsysprof-capture/tests/find-temp-allocs.c: Removed.
* Source/ThirdParty/libsysprof-capture/tests/meson.build: Removed.
* Source/ThirdParty/libsysprof-capture/tests/rewrite-pid.c: Removed.
* Source/ThirdParty/libsysprof-capture/tests/test-capture-cursor.c: Removed.
* Source/ThirdParty/libsysprof-capture/tests/test-capture.c: Removed.
* Source/ThirdParty/libsysprof-capture/tests/test-cplusplus.cpp: Removed.
* Source/ThirdParty/libsysprof-capture/tests/test-mapped-ring-buffer.c: Removed.

Canonical link: https://commits.webkit.org/289297@main
